web/client/plugins/SwipeSettings.jsx

```diff
@@ -8,11 +8,100 @@
 
 import React from 'react';
 import { connect } from 'react-redux';
+import Select from 'react-select';
+import Slider from 'react-nouislider';
+import { createSelector } from 'reselect';
+import {find} from 'lodash';
 
+import Message from '../components/I18N/Message';
+import DockabePanel from '../components/misc/panels/DockablePanel';
+import { layerSwipeSettingsSelector, swipeModeSettingsSelector, spyModeSettingsSelector } from '../selectors/swipe';
+const { setActive, setSwipeToolDirection, setSpyToolRadius } = require('../actions/swipe');
 
-// TODO: UI configuration in TOC
-const MapSwipeSettingsPanel = () => {
-    return (<div id="mapstore-swipe-settings" />);
+const swipeTypeOptions = [{
+    label: 'Horizontal',
+    value: 'cut-horizontal'
+}, {
+    label: 'Vertical',
+    value: 'cut-vertical'
+}];
+
+const SpyRadiusConfiguration = ({spyModeSettings, onSetSpyToolRadius}) => {
+    return (<div className="mapstore-swipe-settings-spy">
+        <h4><Message msgId="toc.radius" /></h4>
+        <div className="mapstore-slider with-tooltip">
+            <Slider
+                tooltips
+                step={1}
+                start={[spyModeSettings.radius]}
+                range={{
+                    'min': [50],
+                    'max': [80]
+                }}
+                onChange={(value) => onSetSpyToolRadius(value[0])}
+            />
+        </div>
+    </div>);
+};
+
+const SwipeTypeConfiguration = ({swipeModeSettings, onSetSwipeToolDirection}) => {
+    return (<div className="mapstore-swipe-settings-slider">
+        <h4><Message msgId="toc.direction" msgParams={{tool: "Swipe"}} /></h4>
+        <Select
+            onChange={({value}) => onSetSwipeToolDirection(value)}
+            clearable={false}
+            value={find(swipeTypeOptions, ['value', swipeModeSettings.direction])}
+            options={swipeTypeOptions}/>
+    </div>);
 };
 
-export default connect(null, {})(MapSwipeSettingsPanel);
+export const SwipeSettings = (
+    {
+        active,
+        configuring,
+        toolMode,
+        swipeModeSettings,
+        spyModeSettings,
+        onSetConfigurationActive = () => {},
+        onSetSpyToolRadius = () => {},
+        onSetSwipeToolDirection = () => {}
+    }) => {
+    return (<div className="mapstore-swipe-settings">
+        <DockabePanel
+            title={toolMode === "spy"
```

Comment:
Please use 2 different labels for this. msgParams are ok for numbers or fixed strings, but in some languages also "splyglass" or "swipe" strings may be localized.